version: "3.8"
services:
  app:
    build:
      context: .
      target: development
    image: auth_server
    environment:
      - PORT_APP=3000
      - SQLALCHEMY_USERNAME:pguser
      - SQLALCHEMY_PASSWORD:pgpassword
      - SQLALCHEMY_DATABASE_NAME:auth_db
      - JWT_SECRET_KEY=super-secret-key
      - TRACING_ENABLED=true
      - TRACING_AGENT_HOST_NAME=jaeger
    ports:
      - "127.0.0.1:3000:3000"
    env_file:
      - ./.env
    volumes:
      - "./app:/src/app"
      - "./migrations:/src/migrations"
    depends_on:
      - redis
      - postgres
      - jaeger

  redis:
    image: redis:6-alpine
    restart: on-failure
    ports:
      - "6379:6379"
    volumes:
      - "redis-data:/data"


  postgres:
    image: postgres:14
    restart: on-failure
    ports:
      - "5432:5432"
    volumes:
      - "postgres-data:/var/lib/postgresql"
    environment:
      POSTGRES_USER: pguser
      POSTGRES_PASSWORD: pgpassword
      POSTGRES_DB: auth_db

  jaeger:
    image: jaegertracing/all-in-one:1.32
    ports:
      - "6831:6831/udp"
      - "5778:5778"
      - "16686:16686"
      - "14250:14250"
      - "14268:14268"
      - "14269:14269"

  nginx:
    image: nginx
    ports:
      - "80:80"
    volumes:
      - ./nginx:/etc/nginx/conf.d
    depends_on:
      - app

volumes:
  redis-data:
    driver: "local"
  postgres-data:
    driver: "local"
